src/components/IdEntry.vue

```diff
@@ -80,39 +79,55 @@
       class="bg-primary hover:bg-primary-hover text-white font-bold shadow-xl uppercase text-lg mx-auto p-4 mt-4 rounded disabled:opacity-50 btn"
       :disabled="isSubmitButtonDisabled"
     >
-      SUBMIT / जमा करें
+      {{ this.textObject["submitButtonText"] }}
     </button>
   </div>
 </template>
 
 <script>
-import { validateSRN } from "@/services/validation.js";
+import { validateID } from "@/services/validation.js";
 import { redirectToDestination } from "@/services/redirectToDestination.js";
 import { sendSQSMessage } from "@/services/API/sqs";
-
-const NUMBER_OF_INPUTS_ALLOWED = 10;
+import { typeToFunctionMap } from "@/services/basicValidationMapping.js";
 const authType = "SRN";
 
 export default {
-  name: "SRNEntry",
+  name: "IdEntry",
   props: {
     redirectTo: String,
     redirectID: String,
     purpose: String,
     purposeParams: String,
+    textObject: Object,
+    inputObject: Object,
+    dataSourceObject: Object,
+    maxNumberOfInput: Number,
+    program: String,
+    authType: String,
   },
   data() {
     return {
       userIDList: [{ userID: "", valid: false }], //array containing user-ids and a valid flag for each
       invalidInputMessage: null, // whether the input is in correct format
       isCurrentUserValid: false, // whether the current user is valid
-      maxLengthOfSRN: 10,
       validateCount: 0, // count the number of times the user has been validated
-      invalidLoginMessage: "Please enter correct SRN / कृपया सही SRN दर्ज करें",
+      invalidLoginMessage: "",
       isLoading: false,
     };
   },
   computed: {
+    /** Returns the input mode stored against the program */
+    inputMode() {
+      return this.inputObject["mode"];
+    },
+    /** Returns the input type stored against the program */
+    inputType() {
+      return this.inputObject["type"];
+    },
```

Comment:
can you give some examples? what is the difference between input type and input mode?